src/main/java/net/neoforged/neoforge/attachment/IAttachmentCloner.java

```diff
@@ -0,0 +1,12 @@
+package net.neoforged.neoforge.attachment;
+
+import javax.annotation.Nullable;
+
+/**
+ * Custom cloner for data attachments, to improve efficiency compared to the default
+ * serialize-copy nbt-deserialize-implementation.
+ */
+public interface IAttachmentCloner<T> {
+    @Nullable
+    T copy(IAttachmentHolder holder, T attachment);
```

Comment:
Please add documentation to this method.

It should notably mention that this function should give "the same" result as the default implementation (i.e. serialize to NBT, then deserialize with the new holder).